common/djangoapps/student/models.py

```diff
@@ -1195,6 +1195,13 @@ def is_enrolled(cls, user, course_key):
         """
         if not user.is_authenticated():
             return False
+
+        # unwrap CCXLocators so that we use the course as the access control
+        # source
+        from ccx_keys.locator import CCXLocator
+        if isinstance(course_key, CCXLocator):
+            course_key = course_key.to_course_locator()
+
```

Comment:
@ormsbee, @pdpinch, @cpennington:

I believe that it is on the radar to completely remove the separate concept of a 'ccx membership' and replace it with an enrollment in a course identified by a `CCXLocator` (which is a [admittedly badly-named] CourseKey).  We have [an issue open](https://github.com/jazkarta/edx-platform/issues/78) in our github tracker for this.

In the interests of limiting the scope of the work to implement custom opaque keys for ccx, we decided to handle that as a separate task.  I agree completely that it should be done, and when it is, this customized code to look up access will be able to be deleted.